src/Aquifer.Common/Services/Caching/CachingVersificationService.cs

```diff
@@ -1,108 +1,216 @@
+        // BookChapters DB table because we use that scheme for Resource verse references (Bible ID 0 has no BibleTexts data).
+        if (bibleId == EngVersificationSchemeBibleId)
+        {
+            return await _memoryCache.GetOrCreateAsync(
+                cacheKey,
+                async cacheEntry =>
+                {
+                    cacheEntry.SlidingExpiration = s_cacheLifetime;
+
+                    return (await _dbContext.BookChapters
+                            .Select(bc => new
+                            {
+                                bc.BookId,
+                                bc.Number,
+                                bc.MaxVerseNumber,
+                            })
+                            .ToListAsync(cancellationToken))
+                        .GroupBy(x => x.BookId)
+                        .ToDictionary(
+                            grp => grp.Key,
+                            grp =>
+                            (
+                                MaxChapterNumber: grp.Max(x => x.Number),
+                                MaxVerseNumberByChapterNumberMap: grp
+                                    .ToDictionary(
+                                        x => x.Number,
+                                        x => x.MaxVerseNumber)
+                                    .AsReadOnly()
+                            ))
+                        .AsReadOnly();
+                })
+                ?? throw new InvalidOperationException($"\"{cacheKey}\" unexpectedly had a null value cached.");
+        }
+
+        // For regular Bibles (not the "eng" versification scheme), get the max verse numbers from the BibleTexts table.
+        return await _memoryCache.GetOrCreateAsync(
+            cacheKey,
+            async cacheEntry =>
+            {
+                cacheEntry.SlidingExpiration = s_cacheLifetime;
 
-        var versificationDictionary = dictionaries?.GetValueOrDefault(bibleId) ??
-                                      new Dictionary<int, int>()
-                                          .AsReadOnly();
-        return versificationDictionary;
+                return (await _dbContext.BibleTexts
+                        .Where(x => x.BibleId == bibleId)
+                        .GroupBy(x => new { x.BookId, x.ChapterNumber })
+                        .Select(grp => new
+                        {
+                            grp.Key.BookId,
+                            grp.Key.ChapterNumber,
+                            MaxVerseNumber = grp.Max(x => x.VerseNumber),
+                        })
+                        .ToListAsync(cancellationToken))
+                    .GroupBy(x => x.BookId)
+                    .ToDictionary(
+                        grp => grp.Key,
+                        grp =>
+                        (
+                            MaxChapterNumber: grp.Max(x => x.ChapterNumber),
```

Comment:
There's an argument that the max chapter number should come from `BibleBookContents` but being that I have to do this grouping to get the max verse number, anyway, I decided to pull it from `BibleTexts` in this single query instead.